src/compiler/checker.ts

```diff
@@ -33711,10 +33727,16 @@ namespace ts {
                 isJSDocTypeAssertion(node);
         }
 
-        function checkDeclarationInitializer(declaration: HasExpressionInitializer, contextualType?: Type | undefined) {
+        function checkDeclarationInitializer(
+            declaration: HasExpressionInitializer,
+            checkMode: CheckMode | undefined,

```

Comment:
Just make this `checkMode: CheckMode` and pass `CheckMode.Normal` instead of `undefined`.